packages/fxa-auth-server/lib/redis.js

```diff
@@ -112,23 +140,40 @@ class FxaRedis extends RedisShared {
    * @param {Buffer | string} tokenId
    */
   async removeRefreshToken(uid, tokenId) {
+    this.metrics?.increment('redis.removeRefreshToken');
+    const span = await tracer.startSpan('redis.removeRefreshToken');
     const p1 = this.redis.hdel(hex(uid), hex(tokenId));
     const p2 = this.resolveInMs(p1, this.timeoutMs);
-    return Promise.race([p1, p2]);
+    const result = await Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
   /**
    *
    * @param {Buffer | string} uid
    */
-  removeRefreshTokensForUser(uid) {
+  async removeRefreshTokensForUser(uid) {
+    this.metrics?.increment('redis.removeRefreshTokensForUser');
+    const span = tracer.startSpan('redis.removeRefreshTokensForUser');
     const p1 = this.redis.unlink(hex(uid));
     const p2 = this.resolveInMs(p1, this.timeoutMs);
-    return Promise.race([p1, p2]);
+    const result = await Promise.race([p1, p2]);
+    span.end();
+    return result;
   }
 
-  get(key) {
-    return this.redis.get(key);
+  async get(key) {
+    this.metrics?.increment('redis.get');
+    const span = tracer.startSpan('redis.get');
+    const result = await this.redis.get(key);
+
+    if (result?.length > 0) {
+      span.setAttribute('redis.get.size', result?.length);
+      this.metrics?.histogram('redis.get.size', result?.length);
```

Comment:
Not sure if these need the `?.` for `result?.length` since this is already wrapped in `if (result?.length > 0) {...}`; and if `result?.length` was `undefined` the `if` above would have failed.